site/content/how-to/monitoring/troubleshooting.md

```diff
@@ -65,54 +65,297 @@ LAST SEEN   TYPE      REASON              OBJECT
+
+To debug NGINX Gateway Fabric, enable verbose logging by editing the `NginxGateway` configuration. This can be done either before or after deploying NGINX Gateway Fabric. Please refer to this [guide](https://docs.nginx.com/nginx-gateway-fabric/how-to/configuration/control-plane-configuration) to do so.
+
 ### NGINX fails to reload
 
 #### Description
 
+NGINX reload errors can occur for various reasons, including syntax errors in configuration files, permission issues, and more. To determine if NGINX has failed to reload, check logs for your _nginx-gateway_ and _nginx_ containers.
+You will see the following error in the _nginx-gateway_  logs `failed to reload NGINX:` followed by the reason for the failure. Similarly, error logs in _nginx_ container start with `emerg`. For example, `2024/06/12 14:25:11 [emerg] 12345#0: open() "/var/run/nginx.pid" failed (13: Permission denied)` shows a critical error, such as a permission problem preventing NGINX from accessing necessary files.
+
+To debug why your reload has failed, start with verifying the syntax of your configuration files by opening a shell in the NGINX container following these [steps](#get-shell-access-to-containers) and running `nginx -T`. If there are errors in your configuration file, the reload will fail and specify why it has failed.
+
+### Understanding the generated config
+
+Understanding the NGINX configuration is key for fixing issues because it shows how NGINX handles requests. This helps tweak settings to make sure NGINX behaves the way you want it to for your application. The configuration file is found at /etc/nginx/nginx.conf within your NGINX container. To understand the usage of NGINX directives in the configuration file, consult this list of [NGINX directives](https://nginx.org/en/docs/dirindex.html).
+
+In this section, we will see how the `nginx.conf` gets updated as we configure different services, deployments and routes with NGINX Gateway Fabric. In the configuration file, you'll often find several server blocks, each assigned to specific ports and server names. NGINX selects the appropriate server for a request and evaluates the URI against the location directives within that block. In cases, where no resources are defined, NGINX Gateway Fabric generates a basic configuration with a default server listening on port 80 for all requests and additional blocks to manage errors with status codes 500 or 502.
+
+This is a default `server` block listening on port 80:
+
+```text
+server {
+    listen 80 default_server;
+
+    default_type text/html;
+    return 404;
+}
+```
+
+Once routes with path matches and rules are defined, the nginx.conf is updated accordingly to determine which location block will manage incoming requests. To demonstrate how `nginx.conf` is changed, lets create some resources:
+
+1. A Gateway with single listener on port 80. The hostname specified is `*.example.com`, so all incoming requests matching that wildcard is accepted by this Gateway.
```

Comment:
This is all great work @salonichf5, but I wonder if we should be getting into this much detail. We will very likely be changing the configuration of these server blocks in the future, and I fear this information will get out of date easily. 

Is it enough to show the user how to get the nginx config? Are all these details necessary @mpstefan @pleshakov @sjberman 

If we do keep the details, I suggest adding a warning that the configuration may change in future releases.

Reply:
I can update it with a warning.

Reply:
@pleshakov @sjberman do you guys have any thoughts on this?

Reply:
Tough... this specific details are really good at showcasing what's going on. I'd add a label for what version you used for the example.

Reply:
Yeah I'm torn. It's good info but also getting close to being a tutorial...it's probably fine with the warning.